Source/Core/DolphinQt/Debugger/BreakpointWidget.cpp

```diff
@@ -161,60 +262,86 @@ void BreakpointWidget::Update()
   if (!isVisible())
     return;
 
-  m_table->clear();
+  const QSignalBlocker blocker(m_table);
 
-  m_table->setHorizontalHeaderLabels(
-      {tr("Active"), tr("Type"), tr("Function"), tr("Address"), tr("Flags"), tr("Condition")});
+  m_table->clear();
+  m_table->setHorizontalHeaderLabels({tr("Active"), tr("Type"), tr("Function"), tr("Address"),
+                                      tr("End Addr"), tr("Break"), tr("Log"), tr("Read"),
+                                      tr("Write"), tr("Condition")});
+  m_table->horizontalHeader()->setStretchLastSection(true);
+
+  // Get row height for icons
+  m_table->setRowCount(1);
+  const int height = m_table->rowHeight(0);
+  m_table->setRowCount(0);
 
-  int i = 0;
-  m_table->setRowCount(i);
+  // Create icon based on row height, downscaled for whitespace padding.
+  const int downscale = static_cast<int>(0.8 * height);
+  QPixmap enabled_icon =
+      Resources::GetThemeIcon("debugger_breakpoint").pixmap(QSize(downscale, downscale));
 
   const auto create_item = [](const QString& string = {}) {
     QTableWidgetItem* item = new QTableWidgetItem(string);
     item->setFlags(Qt::ItemIsEnabled | Qt::ItemIsSelectable);
     return item;
   };
 
+  QTableWidgetItem empty_item = QTableWidgetItem();
+  empty_item.setFlags(Qt::NoItemFlags);
+  QTableWidgetItem icon_item = QTableWidgetItem();
+  icon_item.setData(Qt::DecorationRole, enabled_icon);
+  QTableWidgetItem disabled_item = QTableWidgetItem();
+  disabled_item.setFlags(Qt::NoItemFlags);
+
+  if (Settings::Instance().IsThemeDark())
+    disabled_item.setBackground(QColor(75, 75, 75));
+  else
+    disabled_item.setBackground(QColor(225, 225, 225));
+
   auto& power_pc = m_system.GetPowerPC();
   auto& breakpoints = power_pc.GetBreakPoints();
   auto& memchecks = power_pc.GetMemChecks();
   auto& ppc_symbol_db = power_pc.GetSymbolDB();
 
+  int i = 0;
```

Comment:
Could this be moved or named to something more relevant

Reply:
Guess I shouldn't have moved it from its original position.  I don't want to rename it , since it's used everywhere and not part of my changes.